util/util_test.go

```diff
@@ -108,17 +108,26 @@ func TestGetSubstitutionMap(t *testing.T) {
 		name         string
 		substituData []byte
 		want         map[string]string
+		expectedErr  bool
 	}{
 		{
 			name:         "Test for creating valid map",
 			substituData: []byte(subs),
 			want:         map[string]string{"docker-storage": "/var/lib/docker", "example": "/example/change"},
 		},
+		{
+			name:         "Test for creating map from invalid file",
+			substituData: []byte("GibrishNotYaml"),
+			expectedErr:  true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := GetSubstitutionMap(tt.substituData); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("GetSubstitutionMap() = %v, want %v", got, tt.want)
+			if got, err := GetSubstitutionMap(tt.substituData); !reflect.DeepEqual(got, tt.want) {
+				if err != nil && tt.expectedErr != true {
```

Comment:
```suggestion
				if err != nil && !tt.expectedErr {
```

Testing against true is actually OK in Golang but in some languages it's not safe so I would avoid getting into the habit